code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm

```diff
@@ -370,6 +370,9 @@
 	sac_target.clear_mood_event("shadow_realm")
 	sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
 
+	if(IS_WIZARD(sac_target) || IS_HERETIC(sac_target) || IS_CULTIST(sac_target))
+		sac_target.cure_trauma_type(/datum/brain_trauma/mild/phobia/supernatural,TRAUMA_RESILIENCE_MAGIC)
```

Comment:
instead of applying and then removing the trauma, just don't apply it to begin with